components/mpas-ocean/src/analysis_members/mpas_ocn_eddy_product_variables.F

```diff
@@ -240,66 +258,106 @@ subroutine ocn_compute_eddy_product_variables(domain, timeLevel, err)!{{{
                   velocityMeridionalTimesTemperature_GM(k,iCell) = GMBolusVelocityMeridional(k,iCell)*activeTracers(index_temperature,k,iCell)
+                  velocityZonalTimesSalinity_GM(k,iCell) = GMBolusVelocityZonal(k,iCell)*activeTracers(index_salinity,k,iCell)
+                  velocityMeridionalTimesSalinity_GM(k,iCell) = GMBolusVelocityMeridional(k,iCell)*activeTracers(index_salinity,k,iCell)
                 end do
             end do
             !$omp end do
             !$omp end parallel
+         endif
+
+         ! if submeso is active, include the MLE variables.
+         ! I split the MLE, GM and non-GM variables into separate loops for
+         ! improved readability. 
+         if (config_submesoscale_enable) then
+            call mpas_pool_get_array(eddyProductVariablesAMPool,'velocityZonalTimesTemperature_MLE', velocityZonalTimesTemperature_MLE)
+            call mpas_pool_get_array(eddyProductVariablesAMPool,'velocityMeridionalTimesTemperature_MLE', &
+               velocityMeridionalTimesTemperature_MLE)
+            call mpas_pool_get_array(eddyProductVariablesAMPool,'normalMLEVelocityTimesTemperature', normalMLEVelocityTimesTemperature)
+            call mpas_pool_get_array(eddyProductVariablesAMPool,'velocityZonalTimesSalinity_MLE', velocityZonalTimesSalinity_MLE)
+            call mpas_pool_get_array(eddyProductVariablesAMPool,'velocityMeridionalTimesSalinity_MLE', &
+               velocityMeridionalTimesSalinity_MLE)
+            call mpas_pool_get_array(eddyProductVariablesAMPool,'normalMLEVelocityTimesSalinity', normalMLEVelocityTimesSalinity)
 
-         ! if GM is not active, do exactly the same computation as above but
-         ! without the GM variables.
-         else
             !$omp parallel
             !$omp do schedule(runtime) private(cell1, cell2, k)
             do iEdge = 1,nEdges
                cell1 = cellsOnEdge(1,iEdge)
                cell2 = cellsOnEdge(2,iEdge)
 
                do k = 1, maxLevelEdgeTop(iEdge)
-                  normalVelocityTimesTemperature(k,iEdge) = 0.5_RKIND*(activeTracers(index_temperature,k,cell1) + &
-                               activeTracers(index_temperature,k,cell2)) * normalVelocity(k,iEdge)
-                  normalVelocitySquared(k,iEdge) = normalVelocity(k,iEdge)**2
+                  normalMLEVelocityTimesTemperature(k,iEdge) = 0.5_RKIND*(activeTracers(index_temperature,k,cell1) + &
+                               activeTracers(index_temperature,k,cell2)) * normalMLEVelocity(k,iEdge)
+                  normalMLEVelocityTimesSalinity(k,iEdge) = 0.5_RKIND*(activeTracers(index_salinity,k,cell1) + &
+                               activeTracers(index_salinity,k,cell2)) * normalMLEVelocity(k,iEdge)
                end do
             end do
             !$omp end do
-            !$omp end parallel
 
             !$omp parallel
```

Comment:
@alicebarthel, this line needs to be deleted because each `!$omp parallel` needs to be paired with a `!$omp end parallel`. This one does not have a match. It's best to compile with `OPENMP=true` to catch these. The pairs `!$omp parallel` and  `!$omp end parallel` cat go around several do loops.

Reply:
It is line 313 that needs to be deleted.